plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -312,7 +312,7 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 partitionSpec.specId(),
                 split.getPartitionDataJson(),
                 split.getFileFormat(),
-                split.getSchemaAsJson().map(SchemaParser::fromJson),
+                SchemaParser.fromJson(table.getTableSchemaJson()),
```

Comment:
> Due to internal caching within the method
`org.apache.iceberg.ManifestGroup.planFiles`
the returned file scan tasks may contain an invalid split
schema string.

is it testable?

Reply:
Yes, it it testable through `io.trino.plugin.iceberg.TestIcebergAvroConnectorTest`.

I was reluctant on squashing the two commits of this PR because they address different issues.

The test `io.trino.plugin.iceberg.TestIcebergAvroConnectorTest` covers both of the issues.